dev-tools/release/announcement/announcement.py

```diff
@@ -0,0 +1,237 @@
+email.send_email(msg)
+
+By default, send_email is running in dry mode so it only prints out the generated email.
+
+If you want to send the email, you need to set dry_run=False:
+
+email.send_email(msg, dry_run=False)
+
+Emails are sent to discuss+announcements@elastic.co by default.
+You can set your own destination by setting MAIL_TO env variable.
+You need to define MAIL_SENDER env variable and set it to the sender email address which
+is allowed to send emails to the announcement mailing list.
+
+By default, SMTP server used to send emails is localhost.
+You can define another one by setting SMTP_SERVER env variable.
+"""
+
+import os
+
+env = os.environ
+
+import smtplib
+
+from os.path import dirname, abspath
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+
+from ..logger import logger
+from .. import common
+
+DEV_TOOLS_DIR = common.ROOT_DIR + '/plugin_tools'
+DEFAULT_TEMPLATE_DIR = abspath(os.path.join(abspath(dirname(__file__)), 'templates'))
+USER_TEMPLATE_DIR = common.ROOT_DIR + '/dev-tools/templates'
+
+DEFAULT_EMAIL_RECIPIENT = "discuss%2Bannouncements@elastic.co"
+
+# Generates a Plain/HTML Multipart email from an issue list
+def prepare_email(artifact_id,
+                  release_version,
+                  artifact_name,
+                  artifact_description,
+                  project_url,
+                  issues_bug,
+                  issues_update,
+                  issues_new,
+                  issues_doc):
+
+    total_issues = 0
+    plain_issues_bug = ""
+    html_issues_bug = ""
+    plain_issues_update = ""
+    html_issues_update = ""
+    plain_issues_new = ""
+    html_issues_new = ""
+    plain_issues_doc = ""
+    html_issues_doc = ""
+
+    if issues_bug is not None and len(issues_bug) > 0:
+        # Format content to plain text
+        plain_issues_bug = format_issues_plain(issues_bug, 'Fix')
+        # Format content to html
+        html_issues_bug = format_issues_html(issues_bug, 'Fix')
+        total_issues += len(issues_bug)
+
+    if issues_update is not None and len(issues_update) > 0:
+        # Format content to plain text
+        plain_issues_update = format_issues_plain(issues_update, 'Update')
+        # Format content to html
+        html_issues_update = format_issues_html(issues_update, 'Update')
+        total_issues += len(issues_update)
+
+    if issues_new is not None and len(issues_new) > 0:
+        # Format content to plain text
+        plain_issues_new = format_issues_plain(issues_new, 'New')
+        # Format content to html
+        html_issues_new = format_issues_html(issues_new, 'New')
+        total_issues += len(issues_new)
+
+    if issues_doc is not None and len(issues_doc) > 0:
+        # Format content to plain text
+        plain_issues_doc = format_issues_plain(issues_doc, 'Doc')
+        # Format content to html
+        html_issues_doc = format_issues_html(issues_doc, 'Doc')
+        total_issues += len(issues_doc)
```

Comment:
this section calls for a loop and a dict:

    plain_issues = {}

    for issues in ((issues_bug, 'bug'), (issues_update, 'update'), ...):
        if issues, name: #no need to check for non and length, just check for trueness
            plain_issues[name] = format_issues_plain(issues)
            ....

and then just use the dict when rendering the templates:

    base_context = {
        'release_version': release_version,
        ....
    }
    plain_issues.update(base_contex)
    text = read_email_template('txt') % plain_issues